XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Hands/DefaultHandController.cs

```diff
@@ -0,0 +1,182 @@
+                else
+                {
+                    jointPoses[handJoint] = handData.Joints[i];
+                }
+            }
+
+            //DataProvider.RaiseHandJointsUpdated(InputSource, ControllerHandedness, jointPoses);
+
+            UpdateVelocity();
+            //UpdateInteractions(handData);
+        }
+
+        public override void SetupDefaultInteractions(Handedness controllerHandedness)
+        {
+            AssignControllerMappings(DefaultInteractions);
+        }
+
+        private void UpdateInteractions(SimulatedHandData handData)
+        {
+            lastPointerPose = currentPointerPose;
+            lastGripPose = currentGripPose;
+
+            // For convenience of simulating in Unity Editor, make the ray use the index
+            // finger position instead of knuckle, since the index finger doesn't move when we press.
+            Vector3 pointerPosition = jointPoses[TrackedHandJoint.IndexTip].Position;
+            IsPositionAvailable = IsRotationAvailable = pointerPosition != Vector3.zero;
+
+            if (IsPositionAvailable)
+            {
+                //HandRay.Update(pointerPosition, GetPalmNormal(), CameraCache.Main.transform, ControllerHandedness);
+
+                Ray ray = HandRay.Ray;
+
+                currentPointerPose.Position = ray.origin;
+                currentPointerPose.Rotation = Quaternion.LookRotation(ray.direction);
+
+                currentGripPose = jointPoses[TrackedHandJoint.Palm];
+
+                currentIndexPose = jointPoses[TrackedHandJoint.IndexTip];
+            }
+
+            if (lastGripPose != currentGripPose)
+            {
+                if (IsPositionAvailable && IsRotationAvailable)
+                {
+                    MixedRealityToolkit.InputSystem.RaiseSourcePoseChanged(InputSource, this, currentGripPose);
+                }
+                else if (IsPositionAvailable && !IsRotationAvailable)
+                {
+                    MixedRealityToolkit.InputSystem.RaiseSourcePositionChanged(InputSource, this, currentPointerPosition);
+                }
+                else if (!IsPositionAvailable && IsRotationAvailable)
+                {
+                    MixedRealityToolkit.InputSystem.RaiseSourceRotationChanged(InputSource, this, currentPointerRotation);
+                }
+            }
+
+            for (int i = 0; i < Interactions?.Length; i++)
+            {
+                switch (Interactions[i].InputType)
+                {
+                    case DeviceInputType.SpatialPointer:
+                        Interactions[i].PoseData = currentPointerPose;
+                        if (Interactions[i].Changed)
+                        {
+                            MixedRealityToolkit.InputSystem.RaisePoseInputChanged(InputSource, ControllerHandedness, Interactions[i].MixedRealityInputAction, currentPointerPose);
+                        }
+                        break;
+                    case DeviceInputType.SpatialGrip:
```

Comment:
Its good  you have this, however, in my experience, the spatialgrip is true whenever the pinch is true, on HL2.